### PR TITLE
complete casbinRule and adapter file

### PR DESCRIPTION
**Pull request by lexign**
1. :package: use yarn instead of npm
2. :wrench: add casbinRule model ts file
3. :sparkles: add SequelizeAdapter class ts file

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,32 +4,24 @@
...
   "typings": "lib/adapter.d.ts",
-  "scripts": {
-    "precommit": "lint-staged",
-    "prepublish": "yarn run lint && yarn run test && yarn build",
-    "build": "rimraf lib && tsc",
-    "lint": "tslint \"src/**/*.ts\"",
-    "fix": "tslint \"src/**/*.ts\" --fix",
-    "test": "jest"
-  },
+  "scripts": {},
```
Comment (hsluoyz) [inline package.json]: Why are these deleted?

File: package.json [hunk 2]
```diff
@@ -4,32 +4,24 @@
...
   "devDependencies": {
     "@types/jest": "^23.3.1",
     "@types/lodash": "^4.14.113",
     "@types/node": "^10.5.3",
     "@types/sequelize": "^4.27.24",
+    "coveralls": "^3.0.2",
     "husky": "^0.14.3",
     "jest": "^23.4.2",
     "lint-staged": "^7.2.0",
-    "rimraf": "^2.6.2",
```
Comment (hsluoyz) [inline package.json]: Why is ``rimraf`` deleted?

File: src/adapter.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2018 The Casbin Authors. All Rights Reserved.
```
Comment (hsluoyz) [inline src/adapter.ts]: Please do not modify these.

File: src/adapter.ts [hunk 4]
```diff
@@ -12,155 +12,184 @@
...
-            line.v4 = rule[4];
-        }
-        if (rule.length > 5) {
-            line.v5 = rule[5];
-        }
+  private savePolicyLine(ptype: string, rules: string[]): CasbinRuleAttributes {
+    const line: any = {p_type: ptype};
 
-        return line;
+    if (rules.length > 0) {
```
Comment (hsluoyz) [inline src/adapter.ts]: Do not just copy your code here. I have fixed your typo ``rules`` to ``rule``, and you changed it back again.

File: src/adapter.ts [hunk 5]
```diff
@@ -12,155 +12,184 @@
...
-     * removeFilteredPolicy removes policy rules that match the filter from the storage.
-     */
-    public removeFilteredPolicy(sec: string, ptype: string, rule: string[]) {
-        throw new Error('not implemented');
+    await casbinRule.bulkCreate(lines);
+    return true;
+  }
+
+  // addPolicy adds a policy rule to the storage.
+  public async addPolicy(sec: string, ptype: string, rules: string[]): Promise<void> {
```
Comment (hsluoyz) [inline src/adapter.ts]: Please do not implement ``addPolicy()`` and following functions. Let's make it work first.

File: src/casbinRule.ts [hunk 6]
```diff
@@ -0,0 +1,61 @@
+import * as Sequelize from 'sequelize';
```
Comment (hsluoyz) [inline src/casbinRule.ts]: Please add the copyright header.

**Review by hsluoyz — COMMENTED** [review]
Need to fix.

File: package.json (hunk 2 above)
Comment (lexign) [reply package.json]: This is the lib for fs' rm operating, like `rm -rf {yourPath}`. This repo no need to.

**Comment by hsluoyz** [comment]
The best way is to rework your commits, not submit new commits to fix.

File: src/adapter.ts (hunk 4 above)
Comment (lexign) [reply src/adapter.ts]: Why the variable must be singular? But it's plural at here.

File: src/adapter.ts (hunk 5 above)
Comment (lexign) [reply src/adapter.ts]: I think this function has no problem.

File: src/adapter.ts (hunk 4 above)
Comment (hsluoyz) [reply src/adapter.ts]: In Casbin, a string array like [ptype, v0, v1, v2] is one policy rule. So it's singular.

File: src/adapter.ts (hunk 3 above)
Comment (lexign) [reply src/adapter.ts]: Copy from old repo. I will recover these comment.

File: package.json (hunk 1 above)
Comment (lexign) [reply package.json]: Because when I run the command `yarn install`, it run `jest test` automatically.

File: src/adapter.ts (hunk 5 above)
Comment (hsluoyz) [reply src/adapter.ts]: You just copied your original code here. This makes this new repo not that meaningful if it's just a copy of the old repo. You can notice that the comment is also not the same.

File: package.json (hunk 2 above)
Comment (hsluoyz) [reply package.json]: ok

File: package.json (hunk 1 above)
Comment (hsluoyz) [reply package.json]: I don't understand, if we don't specify ``"test": "jest"``, how do Yarn know which test the project is using? Why not Mocha, why must be jest?

File: src/adapter.ts (hunk 5 above)
Comment (lexign) [reply src/adapter.ts]: Comment or remove?

Comment (hsluoyz) [reply src/adapter.ts]: There are too many other things we need to do first. I still suggest we do not care the optional functions like ``addPolicy()`` for now.

You can keep the code by yourself for now and when others things are done, you can submit it.

Comment (hsluoyz) [reply src/adapter.ts]: Just leave it as before the PR.

File: package.json (hunk 1 above)
Comment (lexign) [reply package.json]: I don't understand, too. But I think the prepublish script run automatically.
```shell
[~] yarn install
yarn install v1.9.4
[1/4] 🔍  Resolving packages...
success Already up-to-date.
$ yarn run lint && yarn run test && yarn build
yarn run v1.9.4
$ tslint "src/**/*.ts"
✨  Done in 2.02s.
yarn run v1.9.4
$ jest
 FAIL  test/adapter.test.ts
  ● Test suite failed to run

    TypeError: sequelize_1.Sequelize.define is not a function

      16 | import { Sequelize } from 'sequelize';
      17 |
    > 18 | const CasbinRule = Sequelize.define('casbin_rule', {
         |                              ^
      19 |     ptype: Sequelize.STRING,
      20 |     v0: Sequelize.STRING,
      21 |     v1: Sequelize.STRING,

      at Object.<anonymous> (src/adapter.ts:18:30)
      at Object.<anonymous> (test/adapter.test.ts:16:1)

Test Suites: 1 failed, 1 total
Tests:       0 total
Snapshots:   0 total
Time:        6.664s
Ran all test suites.
error Command failed with exit code 1.
```

Comment (hsluoyz) [reply package.json]: If you are not sure about a thing, then simply do not change it.
